Sources/XCHammer/XCHammerConfig.swift

```diff
@@ -15,12 +15,59 @@
 import Foundation
 import PathKit
 
-struct XCHammerTargetConfig: Codable {
+/// Xcode Scheme configuration for each target.
+///
+/// @note in some conditions, it's useful to setup different runtime 
+/// configurations for a target with the Xcode GUI. Consider that these options
+/// will not apply to Bazel command line invocations. At the time of writing,
+/// there is no general primitives to express runtime arguments or env in Bazel
+/// in Bazel.
+///
+/// @note this data type _follows_ the format of XcodeGen's Target Scheme
+/// https://github.com/yonaskolb/XcodeGen/blob/master/Docs/ProjectSpec.md#target-scheme
+public struct XCHammerTargetScheme: Codable {
+    public struct ExecutionAction: Codable {
+        public let script: String
+        public let name: String
+        public let settingsTarget: String?
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            script = try container.decode(String.self, forKey: .script)
+            name = try container.decodeIfPresent(String.self, forKey: .name) ?? "Run Script"
+            settingsTarget = try container.decodeIfPresent(String.self, forKey: .settingsTarget)
+        }
+    }
+
+    public struct EnvironmentVariable: Codable, Equatable {
+        public let variable: String
+        public let value: String
+        public let enabled: Bool
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            variable = try container.decode(String.self, forKey: .variable)
+            value = try container.decode(String.self, forKey: .value)
+            enabled = try container.decodeIfPresent(Bool.self, forKey: .enabled) ?? true
+        }
+    }
+
     /// Scheme runtime command line arguments for each target
-    let commandLineArguments: [String]?
-    
+    public let commandLineArguments: [String: Bool]?
+
     /// Scheme runtime environment Variables for each target
-    let environmentVariables: [String: String]?
+    public let environmentVariables: [XCHammerTargetScheme.EnvironmentVariable]?
+
+    /// Xcode Scheme Pre-Actions
+    public let preActions: [ExecutionAction]?
+    
+    /// Xcode Scheme Post-Actions
+    public let postActions: [XCHammerTargetScheme.ExecutionAction]?
+}
+
+struct XCHammerTargetConfig: Codable {
+    /// Meta scheme configuration. Defaults are empty otherwise.
+    let targetScheme: XCHammerTargetScheme?
```

Comment:
This potentially should be moved towards a dictionary. I don't know how to segment scheme configuration by actions like `Build` `Run` `Profile` `Test` otherwise.